qiskit_experiments/curve_analysis/standard_analysis/oscillation.py

```diff
@@ -129,6 +130,7 @@ def _evaluate_quality(self, fit_data: curve.CurveFitResult) -> Union[str, None]:
 
         return "bad"
 
+@deprecated_class("0.5","DumpedOscillationAnalysis is now replaced with DampedOscillationAnalysis")
```

Comment:
This can be used as
```python
from qiskit_experiments.warnings import deprecated_class

class ClassNewName:
    
    def hoge(self):
        print("hoge")
    
@deprecated_class("0.5", new_cls=ClassNewName)
class ClassOldName:
    pass
```

Since `ClassOldName` will be never instantiated, implementation can be removed.

Reply:
@nkanazawa1989, did I do it correctly this time?